packages/@aws-cdk/aws-docdb/lib/parameter-group.ts

```diff
@@ -0,0 +1,86 @@
+import { Construct, IResource, Resource } from '@aws-cdk/core';
```

Comment:
I think we can model the parameter group in a way that is clearer to users. From my own experience, chasing down these parameters can be annoying, and also somewhat error prone. 

Looking at the CloudFormation [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-docdb-dbclusterparametergroup.html#aws-resource-docdb-dbclusterparametergroup--examples), i see three parameters that can be used (there might be more - but just as an example).

```yaml
Parameters: 
  audit_logs: "disabled"
  tls: "enabled"
  ttl_monitor: "enabled"
```   

What if we provided an api like so, on the `DatabaseCluster` construct:

```typescript
cluster.enableAuditLogs()
cluster.enableTls()
cluster.enableTTLMonitor()
```

Internally, these methods will collect the parameters and create and wire the `CfnParameterGroup` on the user's behalf.

We can also have a generic method like:

```typescript
cluster.addParameter(key, value)
```
To support any parameters that we don't have explicit API's for. This will also sets a good example for subsequent contributions, until we eventually cover all these parameters with proper API's, which is heaven. 

It would also be amazing if we could somehow map the `engineVersion` to the `family` of the parameter group, without the user needing to even think about it. 

I know it's not a trivial change, but I think it's not that big also, and provides a great experience. What do you think?

Reply:
I like where you're going with this suggestion, but it does expand the scope of this pull request which focuses on adding support for clusters and instances. Parameter groups are a useful feature of DocumentDB, but are not strictly required to use the service.

As-is, the current API allows users of the `ParameterGroup` L2 construct to supply arbitrary parameters. While the set of supported parameters are not easily discoverable and requires work on their part to find the proper documentation, it is not a show-stopper to being able to use any of the features.

I agree on all your arguments about setting an example and providing a great experience, but would you be opposed to leaving this work to a future PR? I think CDK users would benefit from L2 DocumentDB constructs in their current state and could stand to wait a bit longer for the improved APIs. In the mean-time, I can provide a handy link to the parameter reference documentation in the TSDoc string.

Reply:
@jusiskin I hear you, Lets move forward with the PR as is. I will follow it up with an issue regarding the `ParameterGroup` 👍

Reply:
Thanks. I'm just integrating the rest of your feedback as we speak. Hopefully we can merge this today :)